src/modules/directory.rs

```diff
@@ -161,6 +152,57 @@ fn to_fish_style(pwd_dir_length: usize, dir_string: String, truncated_dir_string
         .join("/")
 }
 
+#[cfg(target_os = "windows")]
+fn to_separator(path: &Path, separator: &str) -> Option<String> {
+    let path_str = path.as_os_str().to_string_lossy();
+
+    let mut out = String::with_capacity((*path_str).len());
+
+    let mut iter = path
+        .iter()
+        .filter(|x| x.to_string_lossy() != "\\" && x.to_string_lossy() != "/");
+    let first = iter.next();
+    if let Some(first_component) = first {
+        out.push_str(&first_component.to_string_lossy());
+        for component in iter {
+            out.push_str(&separator);
+            out.push_str(&component.to_string_lossy());
+        }
+    }
+
+    if path_str.ends_with('\\') || path_str.ends_with('/') {
+        out.push_str(&separator);
+    }
+
+    Some(out)
+}
+
+#[cfg(not(target_os = "windows"))]
+fn to_separator(path: &Path, separator: &str) -> Option<String> {
+    let path_str = path.as_os_str().to_string_lossy();
+
+    if path_str == "/" {
```

Comment:
Maybe use `std::path::MAIN_SEPARATOR` here instead of `/`?

Reply:
Yes, will do. I will have to exclude empty string first and test on len ( std::path::MAIN_SEPARATOR is a char, not a str, remember ;-))